metaflow/client/cache/cache_worker.py

```diff
@@ -0,0 +1,63 @@
+import json
+import importlib
+
+import click
+
+from metaflow.client.cache.cache_action import import_action_class_spec
+
+def best_effort_read(key_paths):
+    for key, path in key_paths:
+        try:
+            with open(path, 'rb') as f:
+                yield key, f.read()
+        except:
+            pass
+
+@click.command()
+@click.option("--request-file",
+              default='request.json',
+              help="Read request from this file.")
+@click.argument('action_spec')
+def cli(action_spec, request_file=None):
+    """
+    Execute an action specified by action_spec.
+    """
+    action_cls = import_action_class_spec(action_spec)
+    with open(request_file) as f:
+        req = json.load(f)
+
+    try:
+        # prepare stream
+        stream = None
+        if req['stream_key']:
+            stream = open(req['stream_key'], 'a', buffering=1)
+            def stream_output(obj):
+                stream.write(json.dumps(obj) + '\n')
+        else:
+            stream_output = None
+
+        # prepare keys
+        keys = list(req['keys'])
+        ex_keys = dict(best_effort_read(req['existing_keys'].items()))
```

Comment:
This seems potentially costly since you assume that all keys will need to be read by the action (you read them already). Also, the keys could be garbage collected so is the contract that this is the value at the time but you shouldn't assume that they will be around when you are done?